src/main/game.cpp

```diff
@@ -81,7 +81,7 @@ void Game::update_piece( GameComponent& component )
   try
   {
     TetrisPiece& piece = dynamic_cast<TetrisPiece&>( component );
-    if( board.has_landed( piece ) && piece.is_falling() )
```

Comment:
Not sure if this is still necessary. However, the shifting state sets a piece's `falling` to false, if it has landed, and prevents it from going into a falling state. As far as I know, it's not hurting anything this way.